Model/FeatureSwitchRepository.php

```diff
@@ -26,6 +26,30 @@ public function getByName($name) {
         return $switch;
     }
 
+    /**
+     * Either creates a new switch or updates an existing switch.
+     *
+     * @param string $name
+     * @param boolean $value
+     * @param boolean $defaultValue
+     * @param int $rolloutPercentage
+     * @return \Bolt\Boltpay\Api\Data\FeatureSwitchInterface
+     */
+    public function upsertByName($name, $value, $defaultValue, $rolloutPercentage) {
+        $switch = null;
+        try {
+            $switch = $this->getByName($name);
+        } catch(\Exception $e) {
```

Comment:
can we be more specific and have NoSuchEntityException here?

Reply:
sure